src/index.js

```diff
@@ -103,13 +122,19 @@ class WebpackDeduplicationPlugin {
             rootPath,
         });
 
-        const dupVals = Object.values(duplicates);
+        const prefixGroups = Object.entries(duplicates);
+        const previousLock = getDedupLock(this.rootPath);
+        const currentLock = {};
 
-        compiler.hooks.normalModuleFactory.tap('WebpackDeduplicationPlugin', (nmf) => {
-            nmf.hooks.beforeResolve.tap('WebpackDeduplicationPlugin', (result) => {
-                return deduplicate(result, dupVals);
+        compiler.hooks.normalModuleFactory.tap(PLUGIN_NAME, (nmf) => {
+            nmf.hooks.beforeResolve.tap(PLUGIN_NAME, (result) => {
+                return deduplicate(result, prefixGroups, previousLock, currentLock);
```

Comment:
Just in terms of API - did you consider turning the "deduplicator" into some sort of instance where you can just pass in the lock as an option - then you can just read the resulting lock from the instance at the end. i.e. something like:

```js
const deduplicator = new Deduplicator({ prefixGroups, lock });
//...
    return deduplicator.deduplicate(result);
///..
writeDedupLock(this.rootPath, deduplicator.getLock());

Reply:
Yep. That looks better.